### ISSUE TITLE
Training loss < 0.01 but LIBERO evaluation success rate remains 0 (using `libero_cogact.py` with Dexmal/libero)

### ISSUE THREAD

**Issue by EmberNoWeither** [dexmal/dexbotic#58]
I trained following the example in `dexbotic/playground/benchmarks/libero/libero_cogact.py` (using 2 nodes with 16 GPUs total, `batch_size=16` per GPU, gradient accumulation steps = 1, `epoch=25`, and keeping all other settings unchanged). The LIBERO dataset was also taken from https://hf-mirror.com/datasets/Dexmal/libero. Although the loss eventually drops below 0.01, evaluating with the final saved weights only yields examples with a success rate of 0.

The model's output:
2026-02-13 11:43:53.331 | INFO     | dexbotic.exp.cogact_exp:_get_response:176 - Processing time: 0.07296610251069069
127.0.0.1 - - [13/Feb/2026 11:43:53] "POST /process_frame HTTP/1.1" 200 -
2026-02-13 11:43:53.750 | INFO     | dexbotic.exp.cogact_exp:_get_response:175 - prompt: <start>A chat between a curious user and an artificial intelligence assistant. The assistant gives helpful, detailed, and polite answers to the user's questions. USER: <image>
What action should the robot take to open the middle drawer of the cabinet? ASSISTANT:  <|endoftext|><end>
action: [[-0.13782279160246258, 0.020529602807232816, -0.15571361494436864, 0.007895286512198924, -0.00046420915333694746, -0.025971041984615528, 0.015336127889156342], [-0.05084400283545265, 0.016448566415423138, -0.13838325007259844, 0.006946077427832847, -0.0057183031466728484, -0.022642343565302037, 0.021157681488990783], [-0.04563872490450749, 0.028069991501240232, -0.16699535559117795, 0.0049392057593119765, -0.0027497337459416182, -0.02287475574952219, 0.02394098945260048], [-0.03439543048292404, -0.001198808355192038, -0.1612623865753412, 0.0050514203493657595, -0.0009116417647113961, -0.0031482741977990614, 0.020757486552000045], [0.1012340188175439, -0.009664501708029238, -0.2172329156845808, 0.022715784127454064, -0.016932606331871825, 0.04512332688508153, 0.10637238634228706], [-0.01988099893927575, 0.0018792394161289971, -0.1545967541858554, 0.0017550709315214963, -0.000503348190883518, -0.003780696227523639, 0.0165456218123436], [-0.008145010326057744, -0.026193662843255083, -0.12465168979391461, 0.0024497306839356414, 0.0012290952643490838, -0.01348330801946962, 0.03732894181013107], [-0.014098299410194248, -0.030461608014995867, -0.14290125380828977, 0.006140627899504342, 0.00533761919161288, -0.008525977288433517, 0.04868972848057747], [-0.013630029190331738, -0.003977855177708545, -0.1369457618370652, 0.009153020240622034, 0.005256681868576557, -0.01318113141613253, 0.02402942505478859], [-0.018672660760581605, 0.023063066360663464, -0.14098313163220888, 0.005442783268425622, -0.0031788628103029404, 0.005988302407321566, 0.00047993000149726867], [0.017569217193871722, 0.03250026057646105, -0.05605685183778408, 0.01090561496812939, -0.0034382667891849628, -0.007974830612866635, 0.29099019787311553], [0.008736114889383306, -0.008307660783716364, -0.1608766726702452, 0.0073632127202968856, -0.002242057852705648, 0.011651682447356149, 0.0], [0.03193748315423717, 0.006696122930747284, -0.09095524481311446, 0.003426755073744042, 0.009128787337743721, 0.015879359658585057, 0.013232564330101013], [-0.057464790422469436, 0.007435793187690831, -0.1665040508285165, 0.004231041702297017, 0.0035994046025461246, 0.005485742359759571, 0.0], [-0.01328517567738896, 0.011590000118916755, -0.18266783226281402, 0.0068882048299934995, 0.002170723645077366, 0.012825778410152089, 0.0], [-0.018299492731690514, 0.02111071919574603, -0.19466893157735465, 0.00989878472371325, -0.013449486599072802, 0.0028008604370660106, 0.0]]
2026-02-13 11:43:53.751 | INFO     | dexbotic.exp.cogact_exp:_get_response:176 - Processing time: 0.07065368071198463

Here is the recorded video during evaluation on **Libero_goal**. Regardless of the task prompt, it always performs the same action.

https://github.com/user-attachments/assets/eaef1375-a3f2-489a-b5c9-84b232f30496

**Comment by EmberNoWeither** [comment]
Here are the final loss logs:
 {
      "epoch": 24.931818181818183,
      "grad_norm": 0.10254672169685364,
      "learning_rate": 3.752510692733857e-10,
      "loss": 0.0059,
      "step": 32910
    },
    {
      "epoch": 24.939393939393938,
      "grad_norm": 0.1394282728433609,
      "learning_rate": 2.9731016347622763e-10,
      "loss": 0.006,
      "step": 32920
    },
    {
      "epoch": 24.946969696969695,
      "grad_norm": 0.16293315589427948,
      "learning_rate": 2.2843198515642984e-10,
      "loss": 0.0073,
      "step": 32930
    },
    {
      "epoch": 24.954545454545453,
      "grad_norm": 0.11219534277915955,
      "learning_rate": 1.6861659673739205e-10,
      "loss": 0.0052,
      "step": 32940
    },
    {
      "epoch": 24.96212121212121,
      "grad_norm": 0.1336306482553482,
      "learning_rate": 1.178640524301944e-10,
      "loss": 0.007,
      "step": 32950
    },
    {
      "epoch": 24.96969696969697,
      "grad_norm": 0.12498496472835541,
      "learning_rate": 7.617439823248696e-11,
      "loss": 0.0057,
      "step": 32960
    },
    {
      "epoch": 24.977272727272727,
      "grad_norm": 0.10021530091762543,
      "learning_rate": 4.354767192626952e-11,
      "loss": 0.0055,
      "step": 32970
    },
    {
      "epoch": 24.984848484848484,
      "grad_norm": 0.18733273446559906,
      "learning_rate": 1.9983903082332333e-11,
      "loss": 0.0076,
      "step": 32980
    },
    {
      "epoch": 24.992424242424242,
      "grad_norm": 0.10006789863109589,
      "learning_rate": 5.483113054705058e-12,
      "loss": 0.0073,
      "step": 32990
    },
    {
      "epoch": 25.0,
      "grad_norm": 0.1479005515575409,
      "learning_rate": 4.5314988428302174e-14,
      "loss": 0.0064,
      "step": 33000
    },
    {
      "epoch": 25.0,
      "step": 33000,
      "total_flos": 6.724400049022409e+19,
      "train_loss": 0.01444400388408791,
      "train_runtime": 55676.1428,
      "train_samples_per_second": 151.689,
      "train_steps_per_second": 0.593
    }

**Comment by yuzhms** [comment]
Could you please check whether the normalization used during training and evaluation is fully consistent? This kind of behavior often indicates that the numerical range of observations or actions becomes abnormal.

Since other official Dexmal checkpoints work normally in your environment, and forward/backward passes run without errors, the issue is unlikely to be caused solely by CUDA 12.8 or PyTorch 2.7.0.

If possible, could you provide more detailed evaluation logs? That would help us better diagnose the issue.
